src/Features/Core/Portable/Shared/Utilities/PatternMatcher.cs

```diff
@@ -37,10 +37,20 @@ private struct Segment
             // respectively.
             public readonly TextChunk[] SubWordTextChunks;
 
-            public Segment(string text, bool verbatimIdentifierPrefixIsWordCharacter)
+            public Segment(string text, bool verbatimIdentifierPrefixIsWordCharacter, bool allowFuzzyMatching)
             {
-                this.TotalTextChunk = new TextChunk(text);
-                this.SubWordTextChunks = BreakPatternIntoTextChunks(text, verbatimIdentifierPrefixIsWordCharacter);
+                this.TotalTextChunk = new TextChunk(text, allowFuzzyMatching);
+                this.SubWordTextChunks = BreakPatternIntoTextChunks(
+                    text, verbatimIdentifierPrefixIsWordCharacter, allowFuzzyMatching);
+            }
+
+            public void Dispose()
```

Comment:
Then we're not as good at pooling objects.  i.e. we'll use a bit more memory.  If you dispose, then we take these objects and give them back to a pool to ensure less churn.